[OpenEXR] add missing underscore for OpenEXR_IEXMATH_LIBRARY_DEBUG NAMES

LaurensVoerman opened the pull request:
fix missing underscore causing the debug library not to be found.
Laurens.

LaurensVoerman committed:
[OpenEXR] add missing underscore for OpenEXR_IEXMATH_LIBRARY_DEBUG NAMES

NancyLi1013 commented:
@LaurensVoerman thanks for this PR.
Could you please also update the `Version `filed in the `CONTROL `file?
I noticed that this port failed on x64-windows, x86-windows, x64-windows-static trilets due to `ogre `and `alembic`.
Could you please take a look and try to fix it ?

LaurensVoerman committed:
updated CONTROL Version; fix AlembicIlmBase.cmake file error caused by debug lib now being found.

LaurensVoerman commented:
I have updated the version number, and added a patch for the cmake error in alembic. The problem with ogre is not changed, this seems to be a conflict between ogre and ogre-next, and not something to do with this PR.
Regards, Laurens.

NancyLi1013 commented:
Thanks for your update and clarify. 
About the conflict between `ogre `and `ogre-next`, could you update [ci.baseline.txt](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) to this PR?
You can add the following lines to `ci.baseline.txt` file.
```
# Conflicts with ogre-next
ogre:x64-windows        = skip
ogre:x64-windows-static = skip
ogre:x86-windows        = skip
```

LaurensVoerman committed:
skip ogre build for all triplets as it conficts with ogre-next

LaurensVoerman commented:
After a failed search for anything actually using IexMath_2_3.dll I wrote an issue anyway:
https://github.com/microsoft/vcpkg/issues/9283
Maybe the issue can be fixed by not building IexMath_2_3 at all, and updating the cmakefiles in dependent projects to not require it, but the current PR seems safer.
Regards, Laurens.

dan-shaw commented on ports/alembic/fix-find-openexr-ilmbase.patch:
```diff
@@ -30,6 +30,18 @@ index cd00d70..a73c8db 100644
+     )
+ 
+-    if (${ALEMBIC_ILMBASE_IEXMATH_LIB})
++    if (ALEMBIC_ILMBASE_IEXMATH_LIB)
```
What does this change do?

LaurensVoerman replied:
Hi Dan,
when the debug .lib cannot be found,
${ALEMBIC_ILMBASE_IEXMATH_LIB} expands to "path/to/IexMath_2_3.lib"
but when the debug .lib is found, it expands to something like:
"optimized;E:/path/to/IexMath-2_3.lib;debug;E:/path/to/IexMath-2_3_d.lib"
Now cmake throws an error:
>CMake Error at CMakeLists.txt:37 (if):
>  if given arguments:
>   "optimized" "E:/path/to/IexMath-2_3.lib" "debug" "E:/path/to/IexMath-2_3_d.lib"
>  Unknown arguments specified
My change causes the IF statement to check the variable to ["be defined to a value that is not a false constant"](https://cmake.org/cmake/help/latest/command/if.html). 
Laurens.

dan-shaw commented:
@LaurensVoerman Thanks for the PR. The `ogre-next` conflict does not seem related to this port. Currently, we are making changes to the MacOS and Windows CI pipeline so some unrelated failures may happen now.

NancyLi1013 commented:
/azp run

NancyLi1013 commented:
@LaurensVoerman 
As some ports depend on `ogre`, we should set all triplets results of `ogre-next` instead of `ogre` to skip in the baseline.
Sorry for my mistake.
Could you please update ci.baseline.txt again?

LaurensVoerman committed:
skip ogre-next instead of skipping ogre

LaurensVoerman commented:
@NancyLi1013 
I have switched to skipping ogre-next in ci.baseline.txt instead of skipping ogre.
searching through all CONTROL files revealed opencv3[ovis] and opencv4[ovis] depend on ogre,
nothing depends on ogre-next.
Laurens.
